### PR TITLE
[WIP] Update Select Component v3

### LINKED ISSUE #179: [BUG] 'Select' component is not styled properly and it is showcased twice
**Describe the bug**
Fix any styling issues with the "Select" component. Guessing the HTML generated from React is not correct, it should just work by applying Bootstrap Italia CSS.

We are also showing **two** "Select" implementations:

- one on the main menu
- one can be found inside "Form" folder

Choose which one we prefer to keep and move/keep it inside "Form" folder, and  remove any unnecessary code.

**Expected behavior**
Please refer to Bootstrap Italia "Select" example: https://italia.github.io/bootstrap-italia/docs/form/select/ and to #169

### PR DESCRIPTION
**Pull request by sanyamdogra**
Fixes #179 

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `next` branch.
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Updates `select` component using `React-select` v3.

#### Changes proposed in this pull request:
Added the following components according to need as mentioned in BI.
- Select Classic
- Select Disabled
- Select Search
- Select Reset
- Select Multi
- Select Group

### CODE REVIEW

**Comment by francescozaia** [comment]
@sanyamdogra I just added a couple of commits to update this branch with latest changes from `next`. Before merging it, we should update and uncomment the **Form** section and any other stories using Select component. Feel free to find any occurrence of the Select component within stories and fix (or remove them if not valid anymore).

**Comment by sanyamdogra** [comment]
Hey @francescozaia, I added a commit PTAL. Regarding Form component, there's still more to it. For that, I was thinking of creating a separate issue and to work upon. Open for suggestions, Thanks.

File: stories/Select/SelectExample.js [hunk 1]
```diff
@@ -1,118 +1,89 @@
-// import React from "react";
-// import PropTypes from "prop-types";
+import React from 'react';
+import Select from 'react-select';
```
Comment (francescozaia) [inline stories/Select/SelectExample.js]: Double quotes here for consistency

File: stories/Select/SelectExample.js [hunk 2]
```diff
@@ -1,118 +1,89 @@
...
-//         label: "Gruppo 1",
-//         options: groupOneOptions
-//     },
-//     {
-//         label: "Gruppo 2",
-//         options: groupTwoOptions
-//     }
-// ];
+    return (
+      <Select
```
Comment (francescozaia) [inline stories/Select/SelectExample.js]: I would suggest to check [this](https://react-select.com/styles) out prior to consider this story complete, in order to style Select component in a way that is more similar to the [UI Kit directives](https://invis.io/WDSQQP543QV#/330541682_italia-DataEntry-Select).

File: stories/Select/SelectExample.js (hunk 1 above)
Comment (sanyamdogra) [reply stories/Select/SelectExample.js]: Done

File: stories/Select/SelectExample.js (hunk 2 above)
Comment (sanyamdogra) [reply stories/Select/SelectExample.js]: Yes, I have been through this earlier. Do you want to style it right now?

**Pull request merged**